internal/lm/resource_test.go

```diff
@@ -216,6 +217,42 @@ func TestGPUResourceLabeler(t *testing.T) {
 
 }
 
+func TestGPUDeviceNameWithParentheses(t *testing.T) {
+	device := resource.DeviceMock{
```

Comment:
Since you also sanitize spaces `sanitised = strings.ReplaceAll(input, " ", "-")` let's add a test for that

Reply:
Sure

Reply:
The reason I didn't add it was because it was already covered in previous test cases

Reply:
We can add:
```
func TestResourceLabeler(t *testing.T) {
	testCases := []struct {
		productParts        []string
		expectedProductName string
	}{
		{
			productParts:        []string{"some", "part"},
			expectedProductName: "some-part",
		},
		{
			productParts:        []string{"some", "", "more other", "parts"},
			expectedProductName: "some-more-other-parts",
		},
	}

	for i, tc := range testCases {
		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
			rl := resourceLabeler{}

			require.EqualValues(t, tc.expectedProductName, rl.getProductName(tc.productParts...))
		})
	}
}
```
to test other combinations and the interactions with the sharing config.

Reply:
I can we can also do a `TrimWhiteSpaces` to further sanitise the string